example/bwmf/bwmf_task.go

```diff
@@ -74,12 +74,150 @@ type bwmfTask struct {
 	childrenReady map[uint64]bool
 
 	dShard, tShard Shard
-	d, t           []*bwmfData
+	d, t           Shard
+
+	// parameters for projected gradient methods
+	sigma, alpha, beta    float64
+	maxIterInSubProblem   int
+	maxIterInFindingAlpha int
 }
 
 // These two function carry out actual optimization.
-func (t *bwmfTask) updateDShard() {}
-func (t *bwmfTask) updateTShard() {}
+func (t *bwmfTask) updateDShard() {
```

Comment:
probably bt or something. t is confusing since it is a common var used by the testing facility of go.